### PR TITLE
fixing opr validation logic + unit tests

### PR DESCRIPTION
**Pull request by WhoSoup**
this is a fix for the points addressed by @PaulBernier in https://github.com/pegnet/pegnet/issues/169

### CODE REVIEW

**Comment by PaulSnow** [comment]
Wow, very good, and good test.

File: opr/grading.go [hunk 1]
```diff
@@ -207,27 +207,30 @@ func GetEntryBlocks(config *config.Config) {
...
+		for _, opr := range oprblocks[i].OPRs { // Go through this block
+			if !VerifyWinners(opr, previousWinners) {
+				continue
+			}
+			opr.Difficulty = opr.ComputeDifficulty(opr.Nonce)
 
+			f := binary.BigEndian.Uint64(opr.SelfReportedDifficulty)
+			if f != opr.Difficulty {
+				// TODO Maybe we should log.warn how many per block are 'malicious'?
+				log.Errorf("Diff mistmatch. Exp %d, found %d", opr.Difficulty, f)
```
Comment (Emyrk) [inline opr/grading.go]: Can we log more context? The ehash and height?

I know I wrote this, but if we are enforcing it, more context would help.
My rational behind the error, was that in the testnet we shouldn't have anyone malicious (unless we mess with ourselves). So if we report this error, there is a bug in our miner/grader that needs addressing. Once we go live, I'm ok with silencing this.

Comment (WhoSoup) [reply opr/grading.go]: i thought about it but the rest of the "self reported difficulty" functionality is not actually in yet. i just wanted to limit this PR to fixing the bugs mentioned, and then someone can build on top of it and implement the self-reported feature properly

Comment (PaulSnow) [reply opr/grading.go]: That's fair

Comment (Emyrk) [reply opr/grading.go]: Ok, that is fine

**Pull request merged**